developer/src/tike/project/Keyman.Developer.System.Project.xmlLdmlProjectFileAction.pas

```diff
@@ -0,0 +1,93 @@
+{
+ * Keyman is copyright (C) SIL International. MIT License.
+ *
+ * xmlLdmlProjectFileAction: actions for LDML keyboard files
+}
+unit Keyman.Developer.System.Project.xmlLdmlProjectFileAction;
+
+interface
+
+uses
+  System.SysUtils,
+
+  Keyman.Developer.System.Project.xmlLdmlProjectFile,
+  Keyman.Developer.System.Project.ProjectFile,
+  Keyman.Developer.System.Project.ProjectFiles,
+  Keyman.Developer.System.Project.ProjectFileType,
+  UKeymanTargets;
+
+type
+  TxmlLdmlProjectFileAction = class(TxmlLdmlProjectFile)
+  private
+    procedure CheckFilenameConventions;
+  public
+    function CompileKeyboard: Boolean;
+    function Clean: Boolean;
+  end;
+
+implementation
+
+uses
+  System.Classes,
+  System.StrUtils,
+  System.Variants,
+  Winapi.Windows,
+
+  Keyman.Developer.System.Project.ProjectLog,
+  Keyman.System.KeyboardUtils,
+  Keyman.Developer.System.KmcWrapper;
+
+function TxmlLdmlProjectFileAction.Clean: Boolean;
+var
+  FJS: string;
+begin
+  CleanFile(OutputFileName);
+
+  if Targets * KMXKeymanTargets <> [] then
+    CleanFile(ExtractFilePath(FileName) + ExtractFileName(ChangeFileExt(KVKFileName, '.kvk')), True);
```

Comment:
Why is KVKFileName used to establish the file name for CleanFile?

Reply:
This is removing a build output, the .kvk file, if it exists.